### PR TITLE
Opentelemetry tags for bound statements

### PR DESCRIPTION
**Pull request by smaczliwka**
The PR extends tracing capabilities using OpenTelemetry standard by implementing OTel tags specific to bound statements. Additional information about the bound statement such as keyspace, table name and partition key is added to the trace.
The implementation was tested with standard tests run by Maven during build, as well as by checking it performance using the added ZipkinUsage example

### CODE REVIEW

File: driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java [hunk 1]
```diff
@@ -182,6 +188,23 @@ public RequestHandler(
...
+      keyspace = ((BoundStatement) statement).getKeyspace();
+
+      ColumnDefinitions boundColumns =
+          ((BoundStatement) statement).statement.getPreparedId().boundValuesMetadata.variables;
+
+      StringBuilder partitionKeyBuilder = new StringBuilder(PARTITION_KEY_MAX_LENGTH);
+      for (int i = 0; i < boundColumns.size(); i++) {
+        String columnName = boundColumns.getName(i);
+        Object value = ((BoundStatement) statement).getObject(i);
+        String valueString = value.toString();
```
Comment (constantfold) [inline driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java]: When `value` is `null`, this crashes with `NullPointerException` as shown in one of integration tests:
```
prepareWithNullValuesTest(com.datastax.driver.core.PreparedStatementTest)  Time elapsed: 0.004 sec  <<< FAILURE!
java.lang.NullPointerException: null
	at com.datastax.driver.core.RequestHandler.<init>(RequestHandler.java:200)
	at com.datastax.driver.core.SessionManager.execute(SessionManager.java:734)
	at com.datastax.driver.core.SessionManager.execute(SessionManager.java:750)
	at com.datastax.driver.core.SessionManager.executeAsync(SessionManager.java:163)
	at com.datastax.driver.core.AbstractSession.execute(AbstractSession.java:58)
	at com.datastax.driver.core.PreparedStatementTest.prepareWithNullValuesTest(PreparedStatementTest.java:427)
```

File: driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java [hunk 2]
```diff
@@ -182,6 +188,23 @@ public RequestHandler(
...
+        partitionKeyBuilder.append(columnName);
+        partitionKeyBuilder.append('=');
+        partitionKeyBuilder.append(valueString);
+        partitionKeyBuilder.append(',');
```
Comment (constantfold) [inline driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java]: There's one `,` too much at the end of generated string.

File: driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java [hunk 3]
```diff
@@ -182,6 +188,23 @@ public RequestHandler(
     } else if (statement instanceof BoundStatement) {
       statementType = "prepared";
       statementText = ((BoundStatement) statement).statement.getQueryString();
+      keyspace = ((BoundStatement) statement).getKeyspace();
+
+      ColumnDefinitions boundColumns =
+          ((BoundStatement) statement).statement.getPreparedId().boundValuesMetadata.variables;
+
+      StringBuilder partitionKeyBuilder = new StringBuilder(PARTITION_KEY_MAX_LENGTH);
+      for (int i = 0; i < boundColumns.size(); i++) {
```
Comment (constantfold) [inline driver-core/src/main/java/com/datastax/driver/core/RequestHandler.java]: Maybe the example I sent was misleading, but the code should not iterate over all bound values (you can bind a value to a column not part of the primary/partition key). Instead it should iterate over only partition key columns - those are alternatively called routing key columns (the partition key determines where to route the request).

The fix is easy - do not iterate over all `boundColumns`, just those specified in `((BoundStatement) statement).statement.getPreparedId().routingKeyIndexes`, i.e.:

``` java
for (int i : ((BoundStatement) statement).statement.getPreparedId().routingKeyIndexes) {
```

**Comment by avelanarius** [comment]
Reiterating my feedback in PR #5: testing, testing, testing. The perfect would be to add tests to both `BasicTracingTest` and `OpenTelemetryTest`, but `BasicTracingTest` is the necessary minimum (additionally a regression test for `NullPointerException` when some column is bound with `null` would be great).

**Review by avelanarius — APPROVED** [review]
LGTM
